vertical-pod-autoscaler/pkg/recommender/model/vpa.go

```diff
@@ -140,6 +154,9 @@ func (vpa *Vpa) UsesAggregation(aggregationKey AggregateStateKey) bool {
 
 // DeleteAggregation deletes aggregation used by this container
 func (vpa *Vpa) DeleteAggregation(aggregationKey AggregateStateKey) {
+	state := vpa.aggregateContainerStates[aggregationKey]
```

Comment:
Why do you clear these two fields if delete is executed just afterwards?

Reply:
Because if the aggregation stops being autoscaled I don't want to keep exporting the diff - the recommendation is bound to get stale, so I need to clear recommendation. I also don't want to mark as recommendation missing, so I clear IsUnderVPA.